src/Illuminate/Database/Eloquent/Collection.php

```diff
@@ -139,6 +139,27 @@ public function map(callable $callback)
     }
 
     /**
+     * Reload a fresh model instance from the database for all the entities.
+     *
+     * @param  array|string  $with
+     * @return static
+     */
+    public function fresh($with = [])
+    {
+        $model = $this->first();
```

Comment:
Sometimes you can't say if Collection will be empty or not. Therefore users have to do it check manually which is wrong.